src/OutputWriters/output_construction.jl

```diff
@@ -16,6 +17,24 @@ end
 ##### Function output fallback
 #####
 
+function mask_field!(output)
+    field_masking_event = mask_immersed_field!(output)
+    wait(field_masking_event)
+end
+
+
+function _construct_output(output, grid, kwargs...)
+    if output isa AbstractOperation
+        output = Field(output)
+    end
+
+    if grid isa ImmersedBoundaryGrid
+        mask_field!(output)
+    end
```

Comment:
@glwagner, would this be better/faster?

```suggestion
    output isa AbstractOperation && output = Field(output)

    grid isa ImmersedBoundaryGrid && mask_field!(output)
```